circuit_knitting/cutting/cut_finding/cutting_actions.py

```diff
@@ -0,0 +1,436 @@
+
+        new_state.add_action(self, gate_spec, (1, w1, rnew))
+
+        return [new_state]
+
+    def export_cuts(
+        self,
+        circuit_interface: SimpleGateList,
+        wire_map: list[Hashable],
+        gate_spec: GateSpec,
+        cut_args,
+    ) -> None:
+        """Insert an LO wire cut into the input circuit for the specified gate and cut arguments."""
+        insert_all_lo_wire_cuts(circuit_interface, wire_map, gate_spec, cut_args)
+
+
+# Add ActionCutLeftWire to the global variable disjoint_subcircuit_actions
+disjoint_subcircuit_actions.define_action(ActionCutLeftWire())
+
+
+def insert_all_lo_wire_cuts(
+    circuit_interface: SimpleGateList,
+    wire_map: list[Hashable],
+    gate_spec: GateSpec,
+    cut_args,
+) -> None:
+    """Insert LO wire cuts into the input circuit for the specified gate and all cut arguments."""
+    gate_ID = gate_spec.instruction_id
+    for input_ID, wire_ID, new_wire_ID in cut_args:
+        circuit_interface.insert_wire_cut(
+            gate_ID, input_ID, wire_map[wire_ID], wire_map[new_wire_ID], "LO"
+        )
+
+
+class ActionCutRightWire(DisjointSearchAction):
+    """Cut the right (second input) wire of a two-qubit gate."""
+
+    def get_name(self) -> str:
+        """Return the look-up name of :class:`ActionCutRightWire`."""
+        return "CutRightWire"
+
+    def get_group_names(self) -> list[str]:
+        """Return the group name of :class:`ActionCutRightWire`."""
+        return ["WireCut", "TwoQubitGates"]
+
+    def next_state_primitive(
+        self,
+        state: DisjointSubcircuitsState,
+        gate_spec: GateSpec,
+        max_width: int,
+    ) -> list[DisjointSubcircuitsState]:
+        """Return the state that results from cutting the right (second input) wire of the gate given by ``gate_spec``."""
+        gate = gate_spec.gate
+
+        # Cutting of multi-qubit gates is not supported in this release.
+        if len(gate.qubits) != 2:  # pragma: no cover
+            raise ValueError(
+                "In this release, only the cutting of two qubit gates is supported."
+            )
+
+        # If the wire-cut limit would be exceeded, return the empty list
+        if not state.can_add_wires(1):
+            return list()
+
+        q1 = gate.qubits[0]
+        q2 = gate.qubits[1]
+        w2 = state.get_wire(q2)
+        r1 = state.find_qubit_root(q1)
+        r2 = state.find_qubit_root(q2)
+
+        if r1 == r2:
+            return list()
+
+        if not state.can_expand_subcircuit(r1, 1, max_width):
+            return list()
+
+        new_state = state.copy()
+
+        rnew = new_state.new_wire(q2)
+        new_state.merge_roots(r1, rnew)
+        new_state.assert_donot_merge_roots(r1, r2)  # Because r1 < rnew
+
+        new_state.gamma_UB = cast(float, new_state.gamma_UB)
+        new_state.bell_pairs = cast(list, new_state.bell_pairs)
+        new_state.bell_pairs.append((r1, r2))
+        new_state.gamma_UB *= 4
```

Comment:
Similar code block to the [above](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/520/files#r1544619261).

Reply:
So this is the square root of the cost (16), right?

Reply:
> So this is the square root of the cost (16), right?

Yes

Reply:
[I am satisfied](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/520#discussion_r1544620733).